File: solutions/ruby/resistor-color-duo/4/resistor_color_duo.rb
module ResistorColorDuo

  BAND = {
    'black'  => 0,
    'brown'  => 1,
    'red'    => 2,
    'orange' => 3,
    'yellow' => 4,
    'green'  => 5,
    'blue'   => 6,
    'violet' => 7,
    'grey'   => 8,
    'white'  => 9
  }

  def self.value(colors)
    colors.take(2).reverse.each_with_index.sum do |color, index|
      BAND[color] * 10**index
    end
  end

end

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(colors: [Color, Color]): number {
  const [first, second] = colors
  return COLORS.indexOf(first) * 10 + COLORS.indexOf(second)
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
] as const

export type Color = (typeof COLORS)[number]
